cake/libs/inflector.php

```diff
@@ -537,7 +537,7 @@ static function underscore($camelCasedWord) {
 	static function humanize($lowerCaseAndUnderscoredWord) {
 		$_this =& Inflector::getInstance();
 		if (!($result = $_this->_cache(__FUNCTION__, $lowerCaseAndUnderscoredWord))) {
-			$result = ucwords(str_replace('_', ' ', $lowerCaseAndUnderscoredWord));
+			$result = $lowerCaseAndUnderscoredWord ? ucwords(str_replace('_', ' ', $lowerCaseAndUnderscoredWord)) : '';
```

Comment:
Shouldn't this check for `null` instead of `false`?
Wouldn't the string `'0'` lead to `$result === ''`, but in the original version to `'0'`?

Reply:
I think the best would be to check if the `$lowerCaseAndUnderscoredWord` is null or not (as the same test is done for throwing deprecated), I shouldnt assume php was checking it that way

Reply:
Sounds good!